### ISSUE TITLE
Implement faster offset/(line,col) conversions

### ISSUE THREAD

**Issue by dragos** [dragos/dragos-vscode-scala#5]
Some call it prefix-sums. Store an array of indexes of the first char of each line for a given file. Use binary search to find the line given an offset. Similarly, use them to get constant-time (line,col)->offset.

**Comment by jvican** [comment]
This is a nice improvement. However, I must say that I'm impressed how fast this plugin is, so not sure if this is really necessary 👿

**Comment by dragos** [comment]
Yeah, that's why it sort of got delayed indefinitely. Still, a nice thing to have :0
